heft-plugins/heft-typescript-plugin/src/TypeScriptPlugin.ts

```diff
@@ -203,7 +203,19 @@ export async function loadPartialTsconfigFileAsync(
           },
           jsonPathMetadata: {
             '$.compilerOptions.outDir': {
-              pathResolutionMethod: PathResolutionMethod.resolvePathRelativeToConfigurationFile
+              pathResolutionMethod: PathResolutionMethod.custom,
+              customResolver(
+                resolverOptions: ConfigurationFile.IJsonPathMetadataResolverOptions<IPartialTsconfig>
+              ): string {
+                if (resolverOptions.propertyValue.includes('${configDir}')) {
```

Comment:
Stick the token in a const.

Reply:
https://github.com/microsoft/rushstack/pull/5444/files#diff-a64df7deb2dc26761033fae0f755466481b865839af3ba7293f36cf0ed399c56R35-R39